### PR TITLE
Fix up tests

### PR DESCRIPTION
**Pull request by mattgodbolt**
* Make intellij test configs use esm and ts-node/register
* make package.json do same
* fix up a couple minor `false` <-> `undefined` json things
* fix logger tests
* fix delphi map reader

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -175,11 +175,11 @@
     "ci-test": "nyc mocha --recursive",
     "lint": "eslint .",
     "lint-fix": "eslint --fix .",
-    "test": "mocha -r esm --recursive",
+    "test": "mocha -r esm -r ts-node/register --recursive",
     "check": "npm run lint && npm run test",
-    "dev": "cross-env NODE_ENV=DEV node -r esm app.js",
-    "debug": "cross-env NODE_ENV=DEV node -r esm app.js --debug",
-    "start": "npx webpack && cross-env NODE_ENV=LOCAL node -r esm app.js",
+    "dev": "cross-env NODE_ENV=DEV node -r esm a-r ts-node/register pp.js",
```
Comment (apmorton) [inline package.json:180]: pp.js

File: lib/map-file-delphi.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
-import { MapFileReader } from './map-file';
+import {MapFileReader} from './map-file';
```
Comment (apmorton) [inline lib/map-file-delphi.ts:25]: I assume this is some automatic intellij refactoring? Can you disable it?

**Comment by mattgodbolt** [comment]
Sorry yeah that's the format I have set. Can and will disable

File: package.json (hunk 1 above)
Comment (mattgodbolt) [reply package.json:180]: Oops: fixed!

File: lib/map-file-delphi.ts (hunk 2 above)
Comment (mattgodbolt) [reply lib/map-file-delphi.ts:25]: Got it: had to copy some settings I had for JS to TS.

**Comment by apmorton** [comment]
manually applied to my branch

**Pull request closed without merging**